### PR TITLE
Airlock Wires For each departments

### PR DESCRIPTION
**Pull request by heepox**
## About The Pull Request

DO NOT MERGE, this is a feature related to an incoming PR
A port of:
https://github.com/tgstation/tgstation/pull/52563

## Why It's Good For The Game

No hacking one door and getting to every single place on the entire station with that knowlage
Insuls once again become the patron saint of assistants everywhere

## Changelog
:cl:Timberpoes, Heepox
add: Department specific randomised wires
balance: Hacking harder now
code: airlocks have "proper_name" (need to get it to show when opening panel (JS stuff PLZ HALP))
/:cl:

### CODE REVIEW

**Comment by DatBoiTim** [comment]
For maintainer information, I am working with Ruko on a skillchip PR, which gives engineers wire reading, among other things.
I was already working on skillchips when Heepox mentioned the idea of porting this, so many hands make for light work.

**Comment by ToasterBan** [comment]
no more gamer wire? both sad and based at the same time, maybe make it a config so lrpers don't shit pant?

**Comment by ItsTisme** [comment]
Merge this after the skillchips merged and then its good. Makes breaking into high secure areas less trivial
(Dont use RCD's instead guys cmon)

**Comment by DatBoiTim** [comment]
> Merge this after the skillchips merged and then its good. Makes breaking into high secure areas less trivial
> (Dont use RCD's instead guys cmon)

This is an established condition for merge, at least between Heepox and I. Maintainers should be able to see the dependence for this and know not to speedmerge this without merging skillchips.

File: code/datums/wires/_wires.dm [hunk 1]
```diff
@@ -40,15 +41,16 @@
 		CRASH("Wire holder is not of the expected type!")
 
 	src.holder = holder
+	var/key = dictionary_key ? dictionary_key : holder_type
```
Comment (park66665) [inline code/datums/wires/_wires.dm]:
```suggestion
	var/key = dictionary_key || holder_type
```
I'd consider `||` here is standard-ish and simpler.

File: code/datums/wires/airlock.dm [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
+	dictionary_key = /datum/wires/airlock/medbay
+	proper_name = "Medbay Airlock"
+
+/datum/wires/airlock/science
+	dictionary_key = /datum/wires/airlock/science
+	proper_name = "Science Airlock"
+
+/datum/wires/airlock/ai
+	dictionary_key = /datum/wires/airlock/ai
+	proper_name = "AI Airlock"
```
Comment (francinum) [inline code/datums/wires/airlock.dm:38]: This should probably be some form of 'High Security' airlock, not necessarily 'ai'

**Comment by TotallyNotChief** [comment]
изображение

File: code/datums/wires/_wires.dm (hunk 1 above)
Comment (heepox) [reply code/datums/wires/_wires.dm]: good idea.

**Comment by PowerfulBacon** [comment]
Stale PR, failing checks. Feel free to reopen when the TGUI code compiles.

**Pull request closed without merging**